### DISCUSSION TITLE
Renovate doesn't update prerelease version to next stable version

### DISCUSSION THREAD

**Discussion by NiGhTTraX** [renovatebot/renovate#19926] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

https://github.com/NiGhTTraX/ts-monorepo/blob/bb80f84b22a7700146640c66944f77fc5944f3f8/apps/cra/package.json#L17 has `"@craco/craco": "~7.0.0-alpha.8"`. Renovate detects that the current version of that dependency is `7.0.0`, but does not create a PR for it. From the [docs](https://docs.renovatebot.com/configuration-options/#ignoreunstable) I expected it to do so:

> if you are on 4.0.0-rc2 and newer versions 4.0.0 and 4.1.0-alpha.1 exist then Renovate will update you to 4.0.0 only

### Relevant debug logs

<details><summary>Logs</summary>

```
          {
            "currentValue": "~7.0.0-alpha.8",
            "currentVersion": "7.0.0",
            "datasource": "npm",
            "depIndex": 8,
            "depName": "@craco/craco",
            "depType": "devDependencies",
            "homepage": "https://craco.js.org",
            "prettyDepType": "devDependency",
            "registryUrl": "https://registry.npmjs.org",
            "sourceDirectory": "packages/craco",
            "sourceUrl": "https://github.com/dilanx/craco",
            "versioning": "npm",
            "warnings": [],
            "updates": []
          },
```

</details>

### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

**Comment by rarkins** [comment, marked as answer]
I've converted to a discussion as this is not a bug (functioning as per our current design) although it may be a feature request, I haven't decided yet.

This is an interesting edge case. The reason there's no upgrade proposed is that `7.0.0` fits into the current range and it doesn't _need_ changing:

![image](https://user-images.githubusercontent.com/6311784/213444801-b020cfa4-7892-4129-9e1b-807ccbbab655.png)

So basically your expectation is that Renovate _narrows_ the existing range so exclude pre-release versions. You do not need to change that range in order to get `7.0.0` - if you ran a fresh `npm install` then you'd get `7.0.0`. The default behavior of rangeStrategy=replace never narrows ranges - only replaces them if a new release doesn't satisfy.

If your expectation is that Renovate instead updates the lock file to `7.0.0`, this is simply a configuration issue, where you can set `rangeStrategy=update-lockfile`.

**Reply by NiGhTTraX** [reply]
Thanks for the quick answer and explanation, @rarkins! I wasn't aware that prerelease versions were treated that way inside ranges. Having learned that, I think it should fall on me to better specify my ranges when using prerelease tags, not on Renovate to support a feature that narrows down ranges to exclude tags, so I will close this. Thanks for the help!
